src/HttpClientPoolItem.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Psr\Http\Message\RequestInterface;
+use Http\Client\Exception;
+
+/**
+ * A HttpClientPoolItem represent a HttpClient inside a Pool.
+ *
+ * It is disabled when a request failed and can be reenable after a certain number of seconds
+ * It also keep tracks of the current number of request the client is currently sending (only usable for async method)
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpClientPoolItem implements HttpClient, HttpAsyncClient
```

Comment:
the class is now exposing many protected methods. this limits flexibility when refactoring. maybe we should keep that all private for now and refactor when you build that persistent solution. you might realize you need a different implementation then - and if they are now private, we can still change everything we need.

probably we need to inject either the strategy or the storage (in-memory, memcache, whatnot) to avoid needing classes for each permutation in-memory+round robin, in-memory+least used, memcache+round robin, ... but i would not work on that yet and just keep things as simple and internal as possible so we can extend later.